### ISSUE TITLE
System messages are no longer displaying

### ISSUE THREAD

**Issue by andybroomfield** [localgovdrupal/localgov_base#154]
Using Localgov_base and themes that derive from it.
Even though the messages block is placed in the messages region, and this is being rendered in the theme, the messages themselves are not being rendered.

Tested by doing a flush cache which should set a status message.
This works in Bartick and Localgov_theme.
Does not show in localgov_base and child theme.

**Comment by andybroomfield** [comment]
Its this section in localgov_base.theme.

````php
foreach ($system_region as $key => $value) {
    $region = $key;
    $has = 'has_' . $region;
    $copy = $variables['page'][$region];
    $rendered = Drupal::service('renderer')->renderPlain($copy);
    $variables[$has] =  (!empty(trim(strip_tags($rendered)))) ? true : false;
  }
````
Removing it brings the messges back (have to force set the has_messages etc variable though).
Is this then marking the section as rendered already so its not triggered, even though it is a copy?

**Comment by andybroomfield** [comment]
Removing the copy variable seems to solve this.
Changing 
````php
 $rendered = Drupal::service('renderer')->renderPlain($copy);
````
to 
````php
 $rendered = Drupal::service('renderer')->renderPlain($variables['page'][$region]);
````
Would this likly effect anything else?

**Comment by andybroomfield** [comment]
Applying the above fix however breaks the map view on a directory channel. 
I guess that is the reason to render it in a copy?

**Comment by markconroy** [comment]
I wonder why this affects the messages block, but not other blocks? And I wonder why your line change fixes the messages block, but breaks (at least) one other block?

**Comment by andybroomfield** [comment]
@markconroy My guess is that the messages block renders a placeholder, so rending the copy might be causing the placeholder for the messages to generate be put into the copy so it won't show. Whereas rending the region itself might be adding in some extra libraries that when re-rendered arn't being attached, as its javascript elements like maps on directories that break.

**Comment by finnlewis** [comment]
Discussing in 'merge monday'. Interesting! @stephen-cox and @finnlewis hope to find some time to investigate this also.

**Comment by markconroy** [comment]
I've made a change here to:

1. Add an array of "special" regions such as messages (currently only messages region is in the array)
2. If an item is in this array, we render it, but not via $copy
3. Added the messages.css file to the global library (it wasn't getting loaded via the template, I think because the template in this case renders too late)

PR #162

**Comment by andybroomfield** [comment]
Can consider fixed?

**Comment by stephen-cox** [comment]
Yep - this was fixed #162
